allennlp/data/data_loaders/multitask_data_loader.py

```diff
@@ -206,56 +173,27 @@ def __init__(
 
     @overrides
     def __len__(self) -> int:
-        if self._instances_per_epoch is not None:
-            return self._instances_per_epoch
-
-        # Here we try to estimate the actual length.  If you are using varying batch size
-        # multipliers per task, we may get batch packing orders that make this an underestimate, as
-        # this assumes that all batches are full, which may not be true.
-        total_instances = 0.0
-        for key, loader in self._loaders.items():
+        if self._instances_per_epoch is None:
             # This will raise a TypeError if any of the underlying loaders doesn't have a length,
-            # which is actually what we want.  If the loader has a length, we set batch_size = 1, so
-            # this will give us the right number of instances.
-            total_instances += self._batch_size_multiplier.get(key, 1.0) * len(loader)
-        if self._drop_last or total_instances % self._batch_size == 0:
-            return int(total_instances) // self._batch_size
+            # which is actually what we want.
+            return self.scheduler.count_batches(
+                {dataset: len(loader) for dataset, loader in self._loaders.items()}
+            )
```

Comment:
Because schedulers now make batches, only schedulers know how many batches they are making. So the responsibility to count is devolved to them.